### PR TITLE
Tag empty state

### PR DESCRIPTION
**Pull request by finben**
Simulator Screen Shot - iPhone 8 - 2020-06-09 at 19 01 36

Here is a screenshot of it in action

### CODE REVIEW

**Comment by finben** [comment]
Simulator Screen Shot - iPhone 8 - 2020-06-09 at 19 02 12

File: Hymns.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -28,7 +28,6 @@
 		3B1B901C2429CF57009FDCC5 /* Assets.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = 3B1B901B2429CF57009FDCC5 /* Assets.xcassets */; };
 		3B1B90222429CF57009FDCC5 /* LaunchScreen.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = 3B1B90202429CF57009FDCC5 /* LaunchScreen.storyboard */; };
 		3B1B90382429CF57009FDCC5 /* HymnsUITests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 3B1B90372429CF57009FDCC5 /* HymnsUITests.swift */; };
-		3B203690248C131B00D90DF5 /* NonFatal.swift in Sources */ = {isa = PBXBuildFile; fileRef = 3B20368F248C131B00D90DF5 /* NonFatal.swift */; };
```
Comment (skywalkerdude) [inline Hymns.xcodeproj/project.pbxproj:31]: Why is this deleted?

File: Hymns.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -222,7 +221,6 @@
 		3B1B90332429CF57009FDCC5 /* HymnsUITests.xctest */ = {isa = PBXFileReference; explicitFileType = wrapper.cfbundle; includeInIndex = 0; path = HymnsUITests.xctest; sourceTree = BUILT_PRODUCTS_DIR; };
 		3B1B90372429CF57009FDCC5 /* HymnsUITests.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = HymnsUITests.swift; sourceTree = "<group>"; };
 		3B1B90392429CF57009FDCC5 /* Info.plist */ = {isa = PBXFileReference; lastKnownFileType = text.plist.xml; path = Info.plist; sourceTree = "<group>"; };
-		3B20368F248C131B00D90DF5 /* NonFatal.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = NonFatal.swift; sourceTree = "<group>"; };
```
Comment (skywalkerdude) [inline Hymns.xcodeproj/project.pbxproj:225]: ditto

File: Hymns/Browse/TagListView.swift [hunk 3]
```diff
@@ -11,11 +11,18 @@ struct TagListView: View {
 
     var body: some View {
         VStack {
+            if viewModel.tags.isEmpty {
+                VStack {
+                    Image("empty tag illustration")
+                    Text("Create tags by tapping on the tag icon on any hymn").padding().multilineTextAlignment(.center)
+                }.frame(minWidth: 0, maxWidth: .infinity, minHeight: 0, maxHeight: .infinity, alignment: .center).background(Color(.systemBackground))
+            } else {
```
Comment (skywalkerdude) [inline Hymns/Browse/TagListView.swift:19]: Nice! Could you create a preview for this state as well? kind of like how I do it here: https://github.com/HymnalDreamTeam/Hymns-iOS/blob/master/Hymns/Display/HymnLyricsView.swift#L56

File: Hymns/Browse/TagListView.swift [hunk 4]
```diff
@@ -11,11 +11,18 @@ struct TagListView: View {
...
             List(viewModel.tags, id: \.self) { tag in
                 NavigationLink(destination: BrowseResultsListView(viewModel: BrowseResultsListViewModel(tag: tag))) {
                     Text(tag)
                 }
             }
+            }
```
Comment (skywalkerdude) [inline Hymns/Browse/TagListView.swift:25]: indent

File: Hymns/Browse/TagListView.swift [hunk 5]
```diff
@@ -11,11 +11,18 @@ struct TagListView: View {
 
     var body: some View {
         VStack {
+            if viewModel.tags.isEmpty {
+                VStack {
+                    Image("empty tag illustration")
+                    Text("Create tags by tapping on the tag icon on any hymn").padding().multilineTextAlignment(.center)
```
Comment (skywalkerdude) [inline Hymns/Browse/TagListView.swift:17]: I talked to Emily about this, but I think it's better to include the tag icon in the message, since it's not super clear from the graphic. can you change this to `Create tags by tapping on the TAG_ICON in any hymn`

File: Hymns/Browse/TagListView.swift [hunk 6]
```diff
@@ -11,11 +11,18 @@ struct TagListView: View {
...
+                }.frame(minWidth: 0, maxWidth: .infinity, minHeight: 0, maxHeight: .infinity, alignment: .center).background(Color(.systemBackground))
```
Comment (skywalkerdude) [inline Hymns/Browse/TagListView.swift:18]: `.maxSize()`

Comment (skywalkerdude) [inline Hymns/Browse/TagListView.swift:18]: Do you need the `.systemBackground`? Is it not that already by default?

File: Hymns/Browse/TagListView.swift (hunk 3 above)
Comment (skywalkerdude) [reply Hymns/Browse/TagListView.swift]: That's possible, but for some reason it works for HymnLyricsView. I tried looking into it, but my previews is broken for some reason and isn't bundling the previews stuff. Can you see why it works for HymnLyricsView but not for this?

File: Hymns/Browse/TagListView.swift (hunk 6 above)
Comment (finben) [reply Hymns/Browse/TagListView.swift]: No, it's transparent. Remember we had this issue before with Chords, Guitar, and Piano. All which where under the Indicator Tab View.

File: Hymns/Browse/TagListView.swift [hunk 7]
```diff
@@ -11,9 +11,20 @@ struct TagListView: View {
...
-                NavigationLink(destination: BrowseResultsListView(viewModel: BrowseResultsListViewModel(tag: tag))) {
-                    Text(tag)
+            if viewModel.tags.isEmpty {
+                VStack(spacing: 5) {
+                    Image("empty tag illustration")
+                    Text("Create tags by tapping on the")
+                    HStack {
+                        Image(systemName: "tag")
+                        Text("icon on any hymn")
+                    }.multilineTextAlignment(.center)
```
[the remark this answers is not in the document]
Comment (finben) [reply Hymns/Browse/TagListView.swift:21]: Yeah, lol that was the first thing I was trying this morning. It never worked. I implemented the whole thing and the image of the SF symbol would just never append even though the front half and back half would append.

File: Hymns/Browse/TagListView.swift (hunk 6 above)
Comment (skywalkerdude) [reply Hymns/Browse/TagListView.swift]: ah, ok. that's fine then

File: Hymns/Browse/TagListView.swift (hunk 3 above)
Comment (skywalkerdude) [reply Hymns/Browse/TagListView.swift]: ok, I figured out why. I found a way to fix it by using .publisher, but we can keep this for now.

File: Hymns/Browse/TagListView.swift (hunk 7 above)
[the remark this answers is not in the document]
Comment (skywalkerdude) [reply Hymns/Browse/TagListView.swift:21]: ok, np. This works then.

File: Hymns/Browse/TagListView.swift (hunk 3 above)
Comment (finben) [reply Hymns/Browse/TagListView.swift]: I was looking at this. I think its because with HymnLyricsView you can feed in the preview to the function since it takes an argument. With TagViewList the function viewModel.getUniqueTags() we can't feed in the results to stub it. Or at least idk how...
        
classic1334ViewModel.lyrics = classic1334ViewModel.convertToViewModels(verses: classic1334_preview.lyrics)

**Comment by skywalkerdude** [comment]
closes https://github.com/HymnalDreamTeam/Hymns-iOS/issues/245

**Comment by finben** [comment]
Is this PR approved?  if so let's get it pushed through the system.
